### DISCUSSION TITLE
Welcome to camera_awesome Discussions!

### DISCUSSION THREAD

**Discussion by g-apparence** [Apparence-io/CamerAwesome#51] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

**Comment by Louixville** [comment]
Hell.o ... I have a question regarding the Sensor type switching for Android ... Is there like a chance to have it for Android in the near future?

**Reply by apalala-dev** [reply]
Supporting this feature on Android is not easily done.
We rely on CameraX, which tries to hide the lens switching feature based on the current camera configuration.
For example: setting the minimal zoom _should_ switch to the ultra wide angle when one is available, since its zoom would be something like x0.6 (less than x1.0).
In reality, many devices are not well supported by CameraX (and sometimes even by Camera2). Some manufacturers just don't let third parties use specific sensors, some do but CameraX don't support it.
This makes it harder to implement, especially since the devs need physical devices with this specific feature supported (mine doesn't 😞).
CameraX has room to grow, hopefully it will be easier in the future but probably not in the _near_ future for sensor type switching.

**Reply by Louixville** [reply]
Thanks for the reply ... Much appreciated.
